tracer/src/Datadog.Trace/Logging/Internal/DatadogLoggingFactory.cs

```diff
@@ -163,13 +164,25 @@ private static string GetDefaultLogDirectory(string? logDirectory)
 #if NETFRAMEWORK
             logDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
 #else
-            if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            var isWindows = FrameworkDescription.Instance.IsWindows();
+
+            ImmutableAzureAppServiceSettings? aasSettings = null;
+            if (source != null)
+            {
+                aasSettings = new ImmutableAzureAppServiceSettings(source, NullConfigurationTelemetry.Instance);
+            }
+
+            if (aasSettings?.IsFunctionsAppConsumptionPlan == true)
```

Comment:
By creating a whole new `ImmutableAzureAppServiceSettings` instance here we're adding a bunch of duplicate work to the start hot-path for _all_ customers (not just aas customers) 🤔 I totally understand why you took this approach, but I wonder if we should instead extract a helper function. 

For example, we're _already_ duplicating some logic for calculating `IsFunctionsAppConsumptionPlan` elsewhere, so we could pull that bit out to a helper specifically, e.g.

```csharp
public class ImmutableAzureAppServiceSettings
{
    public static bool IsFunctionsAppConsumptionPlan(
        IConfigurationSource source,
        IConfigurationTelemetry telemetry)
    {
        var config = new ConfigurationBuilder(source, telemetry);
        var websiteSKU = config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString();
        return IsFunctionsAppConsumptionPlan(websiteSKU);
    }

    private static bool IsFunctionsAppConsumptionPlan(string? websiteSku)
        => websiteSku is "Dynamic" or null;
}
```

You could use those in the `ImmutableAzureAppServiceSettings` when you're building it, but you could also use them directly in `TracerSettings` and here:

```suggestion
            if (ImmutableAzureAppServiceSettings.IsFunctionsAppConsumptionPlan(source , telemetry))
```

(You should pass in `source` and `telemetry` from the callsite for this approach).